src/extra/components/xgm.py

```diff
@@ -0,0 +1,498 @@
+            window_trains (int): The number of trains to use when plotting the
+                rolling average. By default this is chosen automatically based
+                on the number of trains in the run.
+            ax (matplotlib.axes.Axes): The axis to plot in. This will default to
+                [plt][matplotlib.pyplot] if not provided.
+        """
+        intensity = self.intensity(sase)
+
+        if window_trains is None:
+            # Select a number of trains such that the number of points
+            # will be about 500. The goal is to smooth the data just enough
+            # to see something visually useful no matter how long the run is.
+            window_trains = max(1, int(intensity.shape[0] / 500 * 10))
+
+        ax = self._plot_axis_intensity("train", "pulse", sase, ax)
+
+        if window_trains > 1 and len(self._control_source.train_ids) > 300:
+            rolling_mean = intensity.mean(dim="pulseId").rolling(trainId=window_trains, center=True).mean()
+            ax.plot(rolling_mean, label=f"{window_trains / 10:.1f}s rolling average", linewidth=2)
+
+        ax.legend()
+
+        return ax
+
+    def _plot_axis_intensity(self, axis_name, other_axis_name, sase, ax, fmt="-", **kwargs):
+        if ax is None:
+            import matplotlib.pyplot as plt
+            fig, ax = plt.subplots(figsize=(9, 5))
+
+        intensity = self.intensity(sase)
+
+        axis_intensity = intensity.mean(dim=f"{other_axis_name}Id")
+        axis_std = intensity.std(dim=f"{other_axis_name}Id")
+        ax.plot(axis_intensity, fmt, label="Raw data", **kwargs)
+        ax.fill_between(np.arange(len(axis_intensity)),
+                        axis_intensity - axis_std, axis_intensity + axis_std,
+                        alpha=0.5)
+
+        self._set_plot_title(f"Mean {axis_name} intensity (averaged over {other_axis_name}s)", ax, sase)
+        ax.set_xlabel(axis_name.capitalize())
+        ax.set_ylabel("Intensity [μJ]")
+        ax.grid()
+
+        return ax
+
+    def info(self):
+        run_str = ""
+        if self._proposal is not None:
+            run_str = f" for p{self._proposal}, r{self._run_no}"
+
+        try:
+            doocs_server = self.doocs_server()
+        except MultiRunError:
+            doocs_server = "N/A"
+
+        info_str = f"""
+        {self.control_source.source} properties{run_str}:
+          Nominal wavelength:    {self.wavelength():.2f} nm
+          Nominal photon energy: {self.photon_energy():.2f} KeV
+          Avg. pulse energy:     {self.intensity().mean().item():.2f} uJ
+
+          Max pulses:            {self.max_pulses()}
+          Mean pulses:           {self.mean_pulses():.3f}
+
+          DOOCS server:          {doocs_server}
+        """
+
+        print(dedent(info_str))
+
+    def __repr__(self):
+        kev = self.photon_energy()
+        return f"<XGM for {self.control_source.source} with nominal photon energy {kev:.1f} KeV>"
```

Comment:
```suggestion
        return f"<XGM for {self.control_source.source} with nominal photon energy {kev:.1f} keV>"
```

Picky, I know, but the kilo prefix is small k.

Reply:
Oops :facepalm: Fixed in df3e138.